sched/scheduler/workGenerator_test.go

```diff
@@ -0,0 +1,54 @@
+package scheduler
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/queue"
+	"testing"
+)
+
+func TestGenerateWork_ReadsAllItemsFromChannel(t *testing.T) {
```

Comment:
I'm on the fence about this. Dan, would you mind converting one of these to a fake so we can see side-by-side why/if Caitie should eschew the type of testing she's more comfortable/efficient with?